src/Common/Sitemap/Sitemap.php

```diff
@@ -0,0 +1,223 @@
+     * @param (string) $changeFreq changeFreq.
+     *
+     * @since 3.0.0
+     *
+     * @return bool
+    */
+    private function create($mode, $url, $lastMod = null, $priority = 0.5, $changeFreq = 'weekly')
+    {
+        [$lastMod, $priority, $changeFreq] = [$lastMod ?: $this->lastMod, $priority ?: $this->priority, $changeFreq ?: $this->changeFreq];
+        if(!in_array($changeFreq, $this->validFrequencies, true)) {
+            throw new \InvalidArgumentException("The value of changeFreq is not valid", 500);
+        }       
+        $raw = str_replace([':url', ':lastmod', ':changefreq',':priority'], [$url, $lastMod, $changeFreq, (float) $priority], $this->raw);
+        if ($mode === 'create') {
+            $fileH = new SitemapWriter($this->file, 'writeOnly');
+            $fileH->write(Sitemap::START.PHP_EOL);
+            $fileH->write($raw);
+            $fileH->write(PHP_EOL.Sitemap::END);
+        } elseif ($mode === 'append') {
+            $fileH = new SitemapWriter($this->file, 'readOnly');
+            $sitemapData = $fileH->read();
+            $fileH = new SitemapWriter($this->file, 'writeOverride');
+            $sitemapData = str_replace('</urlset>', '', $sitemapData);
+            $sitemapData = $sitemapData . $raw;
+            $fileH->write($sitemapData);
+            $fileH->write(PHP_EOL.Sitemap::END);
+        }
+        $fileH->close();
+
+        return true;
+    }
+
+    /**
+     * Add item to sitemap.
+     *
+     * @param (string) $url        Valid url.
+     * @param (string) $lastMod    Last modify.
+     * @param (float)  $priority   Priority.
+     * @param (string) $changeFreq changeFreq.
+     *
+     * @since 3.0.0
+     *
+     * @return void
+    */
+    public function addItem($url, $lastMod = null, $priority = 0.5, $changeFreq = 'weekly')
+    {
+        if ($this->has($this->file)) {
+            $this->appendItem($url, $lastMod, $priority, $changeFreq);
+        } else {
+            $this->create('create', $url, $lastMod, $priority, $changeFreq);
+        }
+    }
+
+    /**
+     * Append item to sitemap.
+     *
+     * @param (string) $url        Valid url.
+     * @param (string) $lastMod    Last modify.
+     * @param (float)  $priority   Priority.
+     * @param (string) $changeFreq changeFreq.
+     *
+     * @since 3.0.0
+     *
+     * @return void
+    */
+    private function appendItem($url, $lastMod, $priority, $changeFreq)
+    {
+        $this->create('append', $url, $lastMod, $priority, $changeFreq);
+    }
+
+    /**
+     * Determine whether the sitemap exists.
+     **
+     * @since 3.0.0
+     *
+     * @return bool
+    */
+    public function has()
+    {
+        return file_exists($this->file) ? true : false;
+    }
+
+    /**
+     * Delete the sitemap.
+     *
+     * @since 3.0.0
+     *
+     * @return object
+    */
+    public function delete()
+    {
+        if (file_exists($this->file)) {
+            unlink($this->$file);
```

Comment:
This `unlink` will be failed.

The `file_exists` should be true when the `$this->file` is directory.

It should use the `is_dir` and `file_exists` to check it's the file and it's existed.

Reply:
No (Basically) the var `$this->file` is file with path like 

`C:\xampp\htdocs\dev\public/sitemap_index.xml`